### PR TITLE
Added a structure for Octree algorithm

### PR DESCRIPTION
**Pull request by Njord201**
I've added a structure for implement the Octree algorithm

### CODE REVIEW

**Review by mathieurobert1 — APPROVED** [review]
Seems good

File: include/Optimisation/Cube.hpp [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+         * @param minY minY position
+         * @param minZ minZ position
+         * @param maxX maxX position
+         * @param maxY maxY position
+         * @param maxZ maxZ position
+         */
+        Cube(double minX, double minY, double minZ, double maxX, double maxY, double maxZ);
+
+        /**
+         * @brief Construct a new Cube object
```
Comment (AxelF44) [inline include/Optimisation/Cube.hpp:43]:
```suggestion
         * @brief Construct a new Cube object.
```

File: include/Optimisation/Cube.hpp [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+         *
+         */
+        Cube() = default;
+
+        /**
+         * @brief Destroy the Cube object
```
Comment (AxelF44) [inline include/Optimisation/Cube.hpp:49]:
```suggestion
         * @brief Destroy the Cube object.
```

File: src/Plugins/Primitives/Cone/Cone.cpp [hunk 3]
```diff
@@ -142,3 +142,32 @@ Math::Vector3D Primitive::Cone::getNormal(const Math::Vector3D& hitPoint) const
...
+    if (_axis == Axis::X) {
+        collider.maxX.first = true;
+    } else if (_axis == Axis::Y) {
+        collider.maxY.first = true;
+    } else {
+        collider.maxZ.first = true;
+    }
+
+    return collider;
+}
```
Comment (AxelF44) [inline src/Plugins/Primitives/Cone/Cone.cpp:173]:
```suggestion
}

```

**Review by AxelF44 — CHANGES_REQUESTED** [review]

**Review by mathieurobert1 — APPROVED** [review]

**Review by AxelF44 — APPROVED** [review]

**Pull request merged**